kpi/deployment_backends/kobocat_backend.py

```diff
@@ -1070,3 +1066,48 @@ def __validate_bulk_update_submissions(submissions: list) -> list:
             )
         return submissions
 
+    @staticmethod
+    def __prepare_bulk_update_response(kc_responses: list) -> dict:
+        '''
+        Formatting the response to allow for partial successes to be seen
+        more explicitly.
+
+        Args:
+            kc_responses (list): A list containing dictionaries with keys of
+            `_uuid` from the newly generated uuid and `response`, the response
+            object received from Kobocat
+
+        Returns:
+            dict: formatted dict to be passed to a Response object and sent to
+            the client
+        '''
+
+        # Unfortunately, the response message from OpenRosa is in XML format,
+        # so it needs to be parsed before extracting the text
+        results = [
+            {
+                'uuid': res['uuid'],
+                'status_code': res['response'].status_code,
+                'message': ET.fromstring(res['response']._content)
+                .find('{http://openrosa.org/http/response}message')
```

Comment:
what if KoBoCAT really misbehaves and doesn't return XML?
```
In [1]: from xml.etree import ElementTree as ET                                                                                      

In [2]: ET.fromstring('502 Bad Gateway').find('{http://openrosa.org/http/response}message')                                          
Traceback (most recent call last):

  File "/usr/lib/python3/dist-packages/IPython/core/interactiveshell.py", line 3331, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-2-ce5bd0eb196f>", line 1, in <module>
    ET.fromstring('502 Bad Gateway').find('{http://openrosa.org/http/response}message')

  File "/usr/lib/python3.8/xml/etree/ElementTree.py", line 1320, in XML
    parser.feed(text)

  File "<string>", line unknown
ParseError: syntax error: line 1, column 0
```

Reply:
Good point 👍 sorted.